### PR TITLE
CB-10408 modify AwsSubnetIgwExplorer to mark subnet as public when default route table has internet gateway attached to it

### PR DESCRIPTION
**Pull request by biharitomi**
See detailed description in the commit message.

### CODE REVIEW

File: cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorerTest.java [hunk 1]
```diff
@@ -25,19 +29,21 @@
...
-        boolean hasInternetGateway = awsSubnetIgwExplorer.hasInternetGatewayOfSubnet(describeRouteTablesResult, SUBNET_ID);
+        boolean hasInternetGateway = awsSubnetIgwExplorer.hasInternetGatewayOfSubnet(describeRouteTablesResult, SUBNET_ID, VPC_ID);
         assertFalse(hasInternetGateway);
     }
 
     @Test
     public void testWithValidIgw() {
         DescribeRouteTablesResult describeRouteTablesResult = new DescribeRouteTablesResult();
-        RouteTable routeTable = new RouteTable();
+        RouteTable routeTable = new RouteTable().withVpcId(VPC_ID).withVpcId(VPC_ID);
```
Comment (bergerdenes) [inline cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorerTest.java:46]: duplicated

File: cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorerTest.java [hunk 2]
```diff
@@ -83,54 +89,201 @@ public void testWithIgwButNoAssociation() {
...
         describeRouteTablesResult.setRouteTables(List.of(routeTable));
 
-        boolean hasInternetGateway = awsSubnetIgwExplorer.hasInternetGatewayOfSubnet(describeRouteTablesResult, SUBNET_ID);
+        boolean hasInternetGateway = awsSubnetIgwExplorer.hasInternetGatewayOfSubnet(describeRouteTablesResult, SUBNET_ID, VPC_ID);
+
+        assertFalse(hasInternetGateway);
+    }
+
+    @Test
+    public void testWithMainRoutTableAssociationAndNoIgw() {
```
Comment (bergerdenes) [inline cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorerTest.java:143]: spelling: Route

File: cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorerTest.java [hunk 3]
```diff
@@ -83,54 +89,201 @@ public void testWithIgwButNoAssociation() {
...
 
         assertFalse(hasInternetGateway);
     }
 
+    @Test
+    public void testWithMainRoutTableAssociationAndIgw() {
+        DescribeRouteTablesResult describeRouteTablesResult = new DescribeRouteTablesResult();
+        RouteTable routeTable = new RouteTable().withVpcId(VPC_ID);
+        Route route = new Route()
+                .withGatewayId(GATEWAY_ID)
```
Comment (bergerdenes) [inline cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorerTest.java:164]: missing case (could be invalid though):
what if the gateway is there but is not pointing to an open CIDR block

Comment (biharitomi) [reply cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorerTest.java:164]: I think that's covered by this test case: https://github.com/hortonworks/cloudbreak/blob/7d83096e17dd848ee9a3148643830f86660d2253/cloud-aws/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorerTest.java#L66

**Review by bergerdenes — APPROVED** [review]
lgtm

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorer.java [hunk 4]
```diff
@@ -22,43 +23,48 @@
...
+        Set<RouteTable> connectedRouteTables = new HashSet<>();
         for (RouteTable rt : routeTables) {
-            LOGGER.info("Searching the routeTable where routeTable is {} and the subnet is :{}", rt, subnetId);
-            for (RouteTableAssociation association : rt.getAssociations()) {
-                LOGGER.info("Searching the association where association is {} and the subnet is :{}", association, subnetId);
-                if (subnetId.equalsIgnoreCase(association.getSubnetId())) {
-                    LOGGER.info("Found the routeTable which is {} and the subnet is :{}", rt, subnetId);
-                    routeTable = Optional.ofNullable(rt);
-                    break;
+            if (rt.getVpcId().equalsIgnoreCase(vpcId)) {
```
Comment (topolyai5) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorer.java:35]: are there possible to exist route table without the specified vpc? If yes, could you please add a log message or any error if not found with the specific vpc

Comment (biharitomi) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsSubnetIgwExplorer.java:35]: If we consider the entire logic it queries all of the RouteTables for a region at once in one of the upper layers and this part iterates over all of them, I would not like to log those cases.
For example in the engesc that is attached to this ticket the customer has more than 130 route table it seems to be 129 unnecessary log lines to me.

**Review by topolyai5 — APPROVED** [review]

**Pull request merged**